src/layers/microscopy-viewer-layer/data-utils/tiff-utils.js

```diff
@@ -1,50 +1,64 @@
+  return resolvedConnections;
+}
+
+async function loadTile({ image, x, y, pool }) {
   const tile = await image.getTileOrStrip(x, y, 0, pool);
-  const dataObj = {};
   const is8Bits = image.fileDirectory.BitsPerSample[0] === 8;
   const is16Bits = image.fileDirectory.BitsPerSample[0] === 16;
   const is32Bits = image.fileDirectory.BitsPerSample[0] === 32;
-  dataObj[channel] =
+  const data =
     (is8Bits && new Uint8Array(tile.data)) ||
     (is16Bits && new Uint16Array(tile.data)) ||
     (is32Bits && new Uint32Array(tile.data));
-  return dataObj;
+  return data;
 }
 
-export function loadTiff({ connections, x, y, z, pool }) {
-  const configListPromises = Object.keys(connections).map(channel => {
-    const image = connections[channel][z];
-    return loadTile({ image, channel, x, y, pool });
-  });
-  return Promise.all(configListPromises).then(dataList => {
-    const orderedList = [];
-    const dataObj = Object.assign({}, ...dataList);
-    Object.keys(dataObj)
-      .sort()
-      .forEach(key => {
-        orderedList.push(dataObj[key]);
-      });
-    return orderedList;
+export async function loadTiff({ connections, x, y, z, pool }) {
+  const channelPyramids = Object.values(connections);
+  const tileRequests = channelPyramids.map(channelPyramid => {
+    const image = channelPyramid[z];
+    return loadTile({ image, x, y, pool });
   });
+  const tiles = await Promise.all(tileRequests);
+  return tiles;
 }
 
-export function getTiffConnections({ sourceChannels }) {
-  const tiffConnections = Object.keys(sourceChannels).map(async channel => {
-    const tiff = await fromUrl(sourceChannels[channel]);
-    // Get the first image and check its size.
-    const pyramid = await tiff.parseFileDirectories();
-    const minZoom = pyramid.length;
-    const imageObj = {};
-    const imagePromiseList = [];
-    for (let i = 0; i < minZoom; i += 1) {
-      const imagePromise = tiff.getImage(i);
-      imagePromiseList.push(imagePromise);
-    }
-    imageObj[channel] = await Promise.all(imagePromiseList);
-    return imageObj;
-  });
-  return Promise.all(tiffConnections);
+export async function initTiff({ sourceChannels }) {
+  const channelNames = Object.keys(sourceChannels);
+  const channelUrls = Object.values(sourceChannels);
+
+  // Open and resolve all connections asynchronously
+  const tiffConnections = channelUrls.map(url => getChannelConnections(url));
+  const resolvedTiffConnections = await Promise.all(tiffConnections);
+
+  // Map connections to channel keys
+  const connections = {};
+  for (let i = 0; i < channelNames.length; i += 1) {
+    connections[channelNames[i]] = resolvedTiffConnections[i];
+  }
+
+  // Get other properties for viewer
+  const firstFullImage = resolvedTiffConnections[0][0].fileDirectory;
+
+  const minZoom = -1 * resolvedTiffConnections[0].length;
+  const imageWidth = firstFullImage.ImageWidth;
+  const imageHeight = firstFullImage.ImageLength;
```

Comment:
If this is correct, add a little note just to explain to the future that the external and internal names don't match.

Reply:
Will do. I think I'll add this to the issue on creating a more generic OO class for these data functions.